Examples/AWSSDK/Sources/main.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaEvents
+import AWSLambdaRuntime
+import AWSS3
+
+func handler(event: APIGatewayV2Request, context: LambdaContext) async throws -> APIGatewayV2Response {
+
+    var response: APIGatewayV2Response
+    do {
+        // read the list of buckets
+        context.logger.debug("Reading list of buckets")
+        let client = try await S3Client()
```

Comment:
recreating the client over and over looks wasteful. create the client outside the scope of this closure.

Reply:
OK, fixed with a @preconcurrency